master/internal/sproto/task.go

```diff
@@ -166,3 +167,19 @@ type Event struct {
 	// LogEvent is triggered when a new log message is available.
 	LogEvent *string `json:"log_event"`
 }
+
+// GroupID returns the group identifier for an AllocateRequest.
+func (r AllocateRequest) GroupID() *actor.Ref {
+	if r.Group != nil {
+		return r.Group
+	}
+	return r.TaskActor
+}
+
+// UserVisible determines whether the AllocateRequest should
+// be considered in user-visible reports.
+// AKA HasUserVisibleActor.
+func (r AllocateRequest) UserVisible() bool {
+	return r.Group != nil // CHECK we might want to define a new property for this
```

Comment:
yeah i may just make this its own thing. like, a checkpoint gc task for an experiment seems like it should be in the experiment group but not user visible? right? idk, could go either and not make a huge difference but that seems more consistent

Reply:
I agree I went ahead and updated it.